plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.sdk;
+
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+import com.google.cloud.tools.eclipse.sdk.internal.CloudSdkInstallJob;
+import com.google.cloud.tools.managedcloudsdk.ManagedCloudSdk;
+import com.google.cloud.tools.managedcloudsdk.ManagedSdkVerificationException;
+import com.google.cloud.tools.managedcloudsdk.ManagedSdkVersionMismatchException;
+import com.google.cloud.tools.managedcloudsdk.UnsupportedOsException;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponent;
+import java.nio.file.Path;
+
+public class CloudSdkManager {
+
+  /**
+   * Returns {@code CloudSdk} only if there exists an up-to-date Cloud SDK with the App Engine Java
+   * component installed at the managed location. Otherwise, returns {@code null} after triggering a
+   * background installation task. Callers who get {@code null} may need to try this method at later
+   * points.
+   *
+   * Note that even if the Cloud SDK is already installed, this method executes {@code gcloud} twice
```

Comment:
Executing twice on every call seems excessive. I doubt we need to check for updates more than once a day.

Reply:
Yeah, that would reduce two calls to one call. Or, should we also take out `hasComponent()` which calls `gcloud`? I can think of a corner case if we take it out, but I also think there is a way to prevent that.

Regarding checking for the updates once a day, we could

- check for updates whenever Eclipse starts (we could start installing the SDK right after the start-up if it has never been installed, but it may also be a good idea to defer installation)

and

- schedule a job that wakes up every 24 hours and checks for updates.

Reply:
Let's tackle this in a later PR, as it will require a lot of new code and I think how we will tackle this exactly is still to be discussed.